### PR TITLE
Refactor complexity of validation method using string methods extension

### PR DESCRIPTION
**Pull request by RamonMartinezNieto**
Refactor to reduce if concatenation.

It's a very simple thing and maybe unnecessary. I hate concatenating ifs.

Regards.

### CODE REVIEW

File: OpenAI.SDK/OpenAiOptions.cs [hunk 1]
```diff
@@ -164,32 +166,20 @@ private static OpenAiOptions CreateAzureSettingsWithBaseDomain(string apiKey, st
...
+        ApiVersion.ThrowIfIsNullOrEmpty(nameof(ApiVersion));
 
         if (string.IsNullOrEmpty(BaseDomain))
         {
-            if (ProviderType == ProviderType.Azure && string.IsNullOrEmpty(ResourceName))
-            {
-                throw new ArgumentNullException(nameof(ResourceName));
-            }
+            if (ProviderType == ProviderType.Azure)
+                ResourceName?.ThrowIfIsNullOrEmpty(nameof(ResourceName));
```
Comment (kayhantolga) [inline OpenAI.SDK/OpenAiOptions.cs:175]: If ResourceName is null, it won't throw exception.

File: OpenAI.SDK/OpenAiOptions.cs [hunk 2]
```diff
@@ -164,32 +166,20 @@ private static OpenAiOptions CreateAzureSettingsWithBaseDomain(string apiKey, st
...
             throw new ArgumentNullException(nameof(BaseDomain));
         }
 
         if (ProviderType == ProviderType.Azure)
         {
-            if (string.IsNullOrEmpty(DeploymentId))
-            {
-                throw new ArgumentNullException(nameof(DeploymentId));
-            }
+            DeploymentId?.ThrowIfIsNullOrEmpty(nameof(DeploymentId));
```
Comment (kayhantolga) [inline OpenAI.SDK/OpenAiOptions.cs:183]: If DeploymentId is null here, it won't throw exception.

**Comment by kayhantolga** [comment]
Hi Martinez,
Thank you for your contribution. I appreciate your effort and agree that simplifying code is a great idea. However, I noticed some bugs in your code. If you could address these issues, I would be happy to merge your pull request.

**Comment by RamonMartinezNieto** [comment]
Added nullable value type to string (to avoid warnings) and fixed the two bugs.

Sorry for my mistakes and thank you for kindly pointing them out to me. Thank you for your work and effort in this repository, if there is anything I can help with, I'll be happy to do so.

**Comment by kayhantolga** [comment]
Hi again, please try again with these settings. You will see your method will not be triggered when `deploymentId `is null
```
 "OpenAIServiceOptions": {
    "ApiKey": "Put your api key to here or to User Secret file", //How can I get an app key? visit here: https://openai.com/
   // "DeploymentId": "Your Deployment Id",
    "ResourceName": "",
    "ProviderType": "Azure"
    ,"ApiVersion": "Your API Version"//optional
  }
  ```

**Pull request closed without merging**